test/validation/api/pktio/pktio.c

```diff
@@ -2503,6 +2503,245 @@ static void pktio_test_pktout_ts(void)
 	}
 }
 
+static void pktio_test_pktout_compl(bool use_plain_queue)
+{
+	odp_pktio_t pktio[MAX_NUM_IFACES] = {ODP_PKTIO_INVALID};
+	odp_queue_t compl_queue[TX_BATCH_LEN];
+	odp_schedule_capability_t sched_capa;
+	odp_packet_t pkt_tbl[TX_BATCH_LEN];
+	char queuename[ODP_QUEUE_NAME_LEN];
+	odp_pktio_capability_t pktio_capa;
+	odp_queue_capability_t queue_capa;
+	uint16_t seq_found[TX_BATCH_LEN];
+	odp_pktout_queue_t pktout_queue;
+	uint32_t pkt_seq[TX_BATCH_LEN];
+	odp_pktio_t pktio_tx, pktio_rx;
+	odp_packet_tx_compl_t tx_compl;
+	odp_packet_tx_compl_opt_t opt;
+	pktio_info_t pktio_rx_info;
+	odp_pktio_config_t config;
+	odp_queue_param_t qparam;
+	int ret, i, num_rx = 0;
+	odp_event_t ev;
+	uint64_t wait;
+
+	/* Create queues to receive PKTIO Tx completion events */
+	CU_ASSERT_FATAL(!odp_schedule_capability(&sched_capa));
+	CU_ASSERT_FATAL(!odp_queue_capability(&queue_capa));
+
+	for (i = 0; i < TX_BATCH_LEN; i++) {
+		sprintf(queuename, "TxComplQueue%u", i);
+		odp_queue_param_init(&qparam);
+
+		if (use_plain_queue) {
+			qparam.type = ODP_QUEUE_TYPE_PLAIN;
+		} else {
+			qparam.type       = ODP_QUEUE_TYPE_SCHED;
+			qparam.sched.prio = odp_schedule_default_prio();
+			qparam.sched.sync = ODP_SCHED_SYNC_ATOMIC;
+			qparam.sched.group = ODP_SCHED_GROUP_ALL;
+		}
+		compl_queue[i] = odp_queue_create(queuename, &qparam);
+		CU_ASSERT_FATAL(compl_queue[i] != ODP_QUEUE_INVALID);
+	}
+
+	memset(&pktout_queue, 0, sizeof(pktout_queue));
+	CU_ASSERT_FATAL(num_ifaces >= 1);
+
+	/* Open and configure interfaces */
+	for (i = 0; i < num_ifaces; ++i) {
+		pktio[i] = create_pktio(i, ODP_PKTIN_MODE_DIRECT,
+					ODP_PKTOUT_MODE_DIRECT);
+		CU_ASSERT_FATAL(pktio[i] != ODP_PKTIO_INVALID);
+
+		CU_ASSERT_FATAL(odp_pktio_capability(pktio[i], &pktio_capa) == 0);
+
+		/* Configure Tx completion offload for PKTIO Tx */
+		if (i == 0) {
+			CU_ASSERT_FATAL(pktio_capa.tx_compl.mode_all == 1);
+			if (use_plain_queue) {
```

Comment:
I put usually just a dummy comment when checkpatch warns about braces like this, for example:
 "/* CU_ASSERT needs these extra braces */"